src/util.ts

```diff
@@ -0,0 +1,134 @@
+	);
+	console.error(e);
+	res.status(StatusCode.INTERNAL_SERVER_ERROR).json(ServerErrorJSON);
+};
+
+/**
+ * Create a new token for authentication. Requires the
+ * `AUTH_SEED` environment variable to be set to obfuscate
+ * the values of the tokens generated.
+ * @returns the token.
+ */
+export const createToken = (): string => {
+	return createHash("sha256")
+		.update(`${new Date().toISOString()}${process.env.AUTH_SEED}`)
+		.digest("hex");
+};
+
+/**
+ * Returns true if the provided user is authenticated, and false if the
+ * user is not.
+ * @param userID the user's ID as a string.
+ * @param token the string.
+ * @returns a boolean
+ */
+export const isAuthenticated = async (
+	userID: string,
+	token: string | undefined
+): Promise<boolean> => {
+	const userIDBson = new ObjectId(userID);
+
+	try {
+		const user = await getUserByID(userIDBson);
+		if (user.token !== token || !token) {
+			return false;
+		}
+	} catch (e) {
+		if (e instanceof MongoServerError) {
+			console.warn(e);
+		}
+		throw e;
+	}
+	return true;
+};
+
+// following two functions are kinda moist, feel free to DRY it out
+
+export const userExists = async <T>(
+	userID: string,
+	endPointName: string,
+	req: Request<unknown, unknown, T>,
+	res: Response
+): Promise<boolean> => {
+	try {
+		await getUserByID(new ObjectId(userID));
+	} catch (e) {
+		if (e instanceof DbOperationError) {
+			return false;
+		} else {
+			handleError<T>(e, endPointName, req, res);
+		}
+	}
+	return true;
+};
+
+export const placeExists = async <T>(
+	placeID: string,
+	endPointName: string,
+	req: Request<unknown, unknown, T>,
+	res: Response
+): Promise<boolean> => {
+	try {
+		await getPlaceByID(placeID);
+	} catch (e) {
+		if (e instanceof DbOperationError) {
+			return false;
+		} else {
+			handleError<T>(e, endPointName, req, res);
+		}
+	}
+	return true;
+};
+
+// Constants
+export const UserCreatedJSON = { status: "User successfully created." };
+export const RatingCreatedJSON = { status: "Rating successfully created." };
+export const RatingUpdatedJSON = { status: "Rating successfully updated." };
+export const MissingParametersErrorJSON = { error: "This request is missing parameters." };
+export const WrongParamatersErrorJSON = {
+	error: "This request has the wrong data type for a provided field in the body.",
+};
+export const UnauthorizedErrorJSON = {
+	error: "You are not authorized to make this request. Please authenticate and try again.",
+};
+export const ServerErrorJSON = {
+	error: "Something went wrong on the server. Please try again later.",
+};
+export const AccountExistsErrorJSON = { error: "An account already exists with this email." };
+export const AccountNotFoundErrorJSON = {
+	error: "Account with that email and/or password not found.",
+};
+export const RatingAlreadyExistsErrorJSON = {
+	error: "This account has already added a rating to this place.",
+};
+export const PlaceDoesNotExistErrorJSON = {
+	error: "There is no place with the provided place ID.",
+};
```

Comment:
not sure if it's 100% necessary but might it be better to break up this file? for ex, `src/db/Users/util.ts`, `src/rest/Places/util.ts`

Reply:
This has been addressed in my latest changes - I moved the whole util file into the `rest` folder since all of these are only used in the REST endpoint functions. Since `userExists` is clearly related to users and `placeExists` is clearly related to places, I made new `util.ts` files for those functions and added them in their respective folders in `rest` (`rest/Users`, `rest/Places`).